test/e2e_node/e2e_service.go

```diff
@@ -33,22 +34,181 @@ import (
 	"time"
 
 	"github.com/golang/glog"
+	"github.com/kardianos/osext"
 
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+// TODO(random-liu): Move this file to a separate package.
 var serverStartTimeout = flag.Duration("server-start-timeout", time.Second*120, "Time to wait for each server to become healthy.")
 
+// E2EServices starts and stops e2e services. The test use it to start and
+// stop all dependent e2e services.
+type E2EServices struct {
+	cmd *exec.Cmd
+}
+
+func NewE2eServices() *E2EServices {
+	return &E2EServices{}
+}
+
+// StartE2eServices start the e2e services in another process, it returns
+// when all e2e services are ready.
+// We want to statically link e2e services into the test binary, but we don't
+// want their glog to pollute the test result. So we run the binary in start-
```

Comment:
I don't get this. Why can't the test binary start processes in a separate process and then run the tests itself?